awx/api/views/__init__.py

```diff
@@ -404,25 +404,21 @@ class InstanceInstanceGroupsList(InstanceGroupMembershipMixin, SubListCreateAtta
     parent_model = models.Instance
     relationship = 'rampart_groups'
 
-    def unattach_validate(self, request):
-        sub_id, res = super().unattach_validate(request)
-        if res:
-            return (sub_id, res)
-        parent = self.get_parent_object()
-        sub = get_object_or_400(self.model, pk=sub_id)
-        if sub.name == 'controlplane' and parent.node_type == 'hybrid':
-            data = dict({'msg': _(f"Cannot disassociate hybrid instance {parent.hostname} from controlplane.")})
-            res = Response(data, status=status.HTTP_400_BAD_REQUEST)
-            return (sub_id, res)
-        return (sub_id, res)
-
     def is_valid_relation(self, parent, sub, created=False):
         if parent.node_type == 'control':
             return {'msg': _(f"Cannot change instance group membership of control-only node: {parent.hostname}.")}
         if parent.node_type == 'hop':
             return {'msg': _(f"Cannot change instance group membership of hop node : {parent.hostname}.")}
         return None
 
+    def is_valid_removal(self, parent, sub):
+        res = self.is_valid_relation(parent, sub)
+        if res:
+            return res
+        if sub.name == settings.DEFAULT_CONTROL_PLANE_QUEUE_NAME and parent.node_type == 'hybrid':
```

Comment:
I'm pretty sure we don't want to disassociate any instances from the control plane.

Reply:
The other cases are handled in the call to `is_valid_relation` (and I'm not changing the cases in this diff). The only case that isn't covered is "execution"... in which case, I say you should be allowed to remove it.